src/ProjectNeon/Assets/Features/Effects/RepeatOnPrimaryResource.cs

```diff
@@ -0,0 +1,23 @@
+﻿using UnityEngine;
+using System.Collections;
+
+/**
+ * Apply an effect while there is enough Primary Resource
+ */
+public class RepeatOnPrimaryResource : Effect
+{
+    private Effect _effect;
+
+    public RepeatOnPrimaryResource(Effect effect)
+    {
+        _effect = effect;
+    }
+
+    void Effect.Apply(Member source, Target target)
+    {
+        while (source.State[source.State.ResourceTypes[0]]  > 0)
```

Comment:
This bug might be worth creating a test for, to ensure that it doesn't over-expend or get any discounts on repetitions.